tools/migration/cmd/go-filecoin-migrate.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/filecoin-project/go-filecoin/tools/migration/internal"
+	"github.com/filecoin-project/go-filecoin/tools/migration/migrate_1-to-2"
+)
+
+type Migrator interface {
+	Migrate(oldRepo, newRepo *os.File) error
+	Describe()
+	Validate(oldRepo, newRepo *os.File) error
+}
+
+// Run runs the given migration command
+func Run(cmd string, verbose bool) error {
+	var err error
+	miglog, err := makeMigl()
+	logpath := "/tmp/migration_log.txt"
+
+	oldRepo, err := getOldRepo()
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+
+	// open repodir read-only
+	// create new repodir & open read-write
+	// TODO: decide where to put the temp new repo.
```

Comment:
Don't merge TODO without issues. In this case I propose extracting the constant to a field/parameter and requiring it to be specified explicitly for now. My understanding from #2524 is that this will be inferred from the containing directory and involve a timestamp and version number.

One way to go here would be to pass the container `.filecoin` directory here and give this method the responsibility of computing a subdirectory name for the copy.

Reply:
see above